app/templating/schema_context.py

```diff
@@ -43,26 +42,21 @@ def _map_alias_to_answers(aliases, answer_store, answer_ids_on_path):
     return values
 
 
-def _build_exercise(metadata):
+def _build_metadata(metadata):
     return {
-        'start_date': metadata['ref_p_start_date'],
-        'end_date': metadata['ref_p_end_date'],
-        'period_str': json_and_html_safe(metadata['period_str']),
-        'employment_date': metadata['employment_date'],
-        'return_by': metadata['return_by'],
-        'region_code': json_and_html_safe(metadata['region_code']),
-    }
-
-
-def _build_respondent(metadata):
-    return {
-        'ru_name': json_and_html_safe(metadata['ru_name']),
-        'trad_as': json_and_html_safe(metadata['trad_as']),
-        'trad_as_or_ru_name': json_and_html_safe(metadata['trad_as'] or metadata['ru_name']),
+        'ref_p_start_date': metadata.get('ref_p_start_date'),
```

Comment:
Same here - we should be able to build this up from the `required_metadata` in the schemas.

I don't think we should need to change the code to accept new schema-specific metadata. Ideally we should just be able to add the key to the `required_metadata`.

Reply:
For the schema_context, I'll just iterate through the `required_meatadata` as there are no eq specific keys here.